### PR TITLE
✨[api/gcpmachine] Support root volume type param.

### PR DESCRIPTION
**Pull request by dilyevsky**
**What this PR does / why we need it**:

Support confirurable root volume type as (optional) parameter within
GCPMachineSpec.

Supported types:
* `pd-standard` - Standard persistent disk (default)
* `pd-ssd` - SSD persistent disk

The above are only disk types currently supported by GCP as root volumes.

### CODE REVIEW

**Comment by dilyevsky** [comment]
@detiber @justinsb friendly ping

File: api/v1alpha3/gcpmachine_types.go [hunk 1]
```diff
@@ -77,6 +84,14 @@ type GCPMachineSpec struct {
 	// +optional
 	RootDeviceSize int64 `json:"rootDeviceSize,omitempty"`
 
+	// RootDeviceType is the type of the root volume.
+	// Supported types of root volumes:
+	// 1. "pd-standard" - Standrad (HDD) persistent disk
```
Comment (rsmitty) [inline api/v1alpha3/gcpmachine_types.go:89]: small typo here: "Standrad". looks like it also got copy/pasted in a few other places below.

**Comment by rsmitty** [comment]
May be that this needs a rebase to pull in https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/309

File: api/v1alpha3/gcpmachine_types.go (hunk 1 above)
Comment (dilyevsky) [reply api/v1alpha3/gcpmachine_types.go]: Thanks, fixed!

**Review by dilyevsky — COMMENTED** [review]
Hey @rsmitty thanks for review, rebased and fixed typo. PTAL

File: config/crd/bases/infrastructure.cluster.x-k8s.io_gcpmachines.yaml [hunk 2]
```diff
@@ -101,6 +101,11 @@ spec:
                   Defaults to 30.
                 format: int64
                 type: integer
+              rootDeviceType:
+                description: 'RootDeviceType is the type of the root volume. Supported
+                  types of root volumes: 1. "pd-standard" - Standrad (HDD) persistent
```
Comment (rsmitty) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_gcpmachines.yaml:106]: Still a typo here and in config/crd/bases/infrastructure.cluster.x-k8s.io_gcpmachinetemplates.yaml

Comment (dilyevsky) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_gcpmachines.yaml]: Doh, forgot to rerun `make generate`. Fixed (looks like it also picked up unrelated doc string change that was made previously.

**Pull request merged**